src/Core/Settings.h

```diff
@@ -139,6 +139,7 @@ class IColumn;
     \
     M(UInt64, min_bytes_to_use_direct_io, 0, "The minimum number of bytes for reading the data with O_DIRECT option during SELECT queries execution. 0 - disabled.", 0) \
     M(UInt64, min_bytes_to_use_mmap_io, 0, "The minimum number of bytes for reading the data with mmap option during SELECT queries execution. 0 - disabled.", 0) \
+    M(Bool, checksum_on_read, true, "Validate checksums on reading. It is enabled by default and should be always enabled in production. Please do not expect any benefits in disabling this setting. It may only be used for experiments and benchmarks. The setting only applicable for tables of MergeTree family. Checksums are always validated for other table engines and when receiving data over network.", 0) \
```

Comment:
Maybe make this name longer (and less convenient)? `merge_tree_disable_checksum_on_read`?

Reply:
We have settings like `min_bytes_to_use_mmap_io` that are also applied only for MergeTree but they can be extended for Log-engines. I wish to treat `checksum_on_read` in the same way.